### ISSUE TITLE
FPFile array empty on onActivityResult

### ISSUE THREAD

**Issue by mithepner** [filestack/filepicker-android#101]
I'm not able to reproduce this issue myself but some users are experiencing it.  Sometimes we get an IndexOutOfBoundsException because the FPFile array is empty when we return from the FilePicker activity.  My onActivityResult() method is pretty much the same as in the getting started guide.

```
@Override
public void onActivityResult(int requestCode, int resultCode, Intent data) {
    if (requestCode == Filepicker.REQUEST_CODE_GETFILE) {
        if (resultCode != Activity.RESULT_OK) {
            //Result was cancelled by the user or there was an error
            return;
        }

        // Filepicker always returns array of FPFile objects
        ArrayList<FPFile> fpFiles = data.getParcelableArrayListExtra(Filepicker.FPFILES_EXTRA);

        // Get first object (use if multiple option is not set)
        FPFile file = fpFiles.get(0);  // <- java.lang.IndexOutOfBoundsException: Invalid index 0, size is 0

    }
}
```

This happened on an HTC One running android4.4.2

Thanks

**Comment by jschnall** [comment]
Seeing a similar issue on my Nexus 6.  It happens every time I try to grab an image from the camera or gallery.

**Comment by mithepner** [comment]
interesting.  What kind of phone do you have?  I can't reproduce it on any of our devices.  I only see it in testing.

**Comment by mithepner** [comment]
Looks like they're not working on this.  I just contacted support and got the following response.
	
Sarajane Stevens (Filestack)
Aug 2, 14:18 CST

Hi Tim,

Unfortunately our resources do not allow us to investigate this further currently. We hope to add an Android Developer to our team who can address these issues in the near future.

Thank you for your patience,
Sarajane

**Comment by jschnall** [comment]
Nexus 5X, running Android 6.0.1.  It's one of the  current Google reference
devices, so I definitely wouldn't expect issues.

Joel

**Comment by janhudzik** [comment]
Unfortunately I didn't manage to reproduce this issue. I have tried on Nexus 5X (Android 7.0), Samsung Galaxy S5 (Android 6.0.1) and Samsung Galaxy S4 Mini (Android 4.4.2). 

Nevertheless I have posted a fix (087e000b753574f5e21446a62dced84df44d1072) which should help. It makes sure that successful get file(s) operation always returns a non-empty list of the FPFile objects. It will be availabe in the 3.10.0 release.

If you ever enounter this issue again please reopen this and provide all possible details (Android version, device logs with stacktrace, snippet from the code you use to retrieve this list).
